algorithms/kernel/dtrees/dtrees_feature_type_helper.i

```diff
@@ -193,7 +193,7 @@ template <typename IndexType, typename algorithmFPType, CpuType cpu>
 services::Status ColIndexTaskBins<IndexType, algorithmFPType, cpu>::makeIndex(NumericTable & nt, IndexedFeatures::FeatureEntry & entry,
                                                                               IndexType * aRes, size_t iCol, size_t nRows, bool bUnorderedFeature)
 {
-    if (bUnorderedFeature || nRows <= _prm.maxBins * _prm.minBinSize) return this->makeIndexDefault(nt, entry, aRes, iCol, nRows, bUnorderedFeature);
```

Comment:
Check:
1) correctness
2) case for a few unique values

Reply:
1. This change moves algorithm to its expected behavior. Real max number of bins will be equal to number of rows (may be greater than maxBins parameter, which is unacceptable) without it.
2. Few unique values are correctly handled in makeIndexDefault method